src/datasets/base.py

```diff
@@ -3,7 +3,54 @@
 A base class for mobility datasets.
 """
 import abc
-import os, pandas as pd
+import logging
+from pathlib import Path
+import pandas as pd
+from sklearn.model_selection import StratifiedShuffleSplit
+
+
+from src import config
+
+LOG = logging.getLogger(__name__)
+
+
+def stratified_split(
+    df: pd.DataFrame,
+    label_column: str,
+    trajectory_column: str,
+    test_size: float = 0.2,
+    min_trajectories: int = 10,
+):
+    """Split the dataset into train and test sets, stratifying trajectories by label.
+
+    A stratified split on label to balance label classes across the
+    test and train sets. Random assignment is at the group (trajectory
+    ID) level.  Labels with less than 2 trajectories will be dropped
+    so that each label shows up at least once in each of the train and
+    test sets.
+
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        A pandas DataFrame containing the trajectory data to split.
+    label_column: str
+        The name of the column containing the label to stratify by.
+    trajectory_column: str
+        The name of the column containing the trajectory ID to group assign by.
+    test_size : float
+        The ratio of the data that should be assigned to the test set. Default is 20%.
+    min_trajectories : int
+        The minimum number of trajectories a subject must have in order to be included.
+    """
+    ids = df[[label_column, trajectory_column]].drop_duplicates()
+    ids = ids.groupby("label").filter(lambda x: len(x) >= min_trajectories)
+    split = StratifiedShuffleSplit(test_size=test_size, n_splits=2, random_state=config.SEED).split(
```

Comment:
Cool function! Seems like a great approach for keeping consistent sample sizes, but SEED is always 11?

Reply:
I see that it is overwritten in mobility_cli.py when run